src/nvim/lua/vim.lua

```diff
@@ -93,6 +93,21 @@ local function _os_proc_children(ppid)
   return children
 end
 
+-- Default paste function.
+local function _paste(lines)
+  -- local eof = (lines == {''})
+  local call = vim.api.nvim_call_function
+  local mode = call('mode', {})
+  local curline = call('line', {'.'})
+  -- vim.api.nvim_set_option('paste', true)
+  vim.api.nvim_put(lines, 'c', false)
+  -- vim.api.nvim_set_option('paste', false)
+  -- TODO: do not redraw (slow!) until paste is finished.
+  -- if eof then
+  vim.api.nvim_command('redraw')
```

Comment:
should not be needed (perhaps `nvim_put` lack a call to mark buffer as changed?)

Reply:
Doesn't seem so, `do_put` already calls `changed_lines` (which schedules any window containing the buffer to be redrawn already).